UnityProject/Assets/Scripts/Items/Weapons/Gun.cs

```diff
@@ -562,6 +562,21 @@ public virtual bool WillInteract(AimApply interaction, NetworkSide side)
 			return false;
 		}
 
+		//.magnitude leaves out small area in the outer corner of tiles diagonal to the player pos
+		//Doing it this way prevents that
+		private bool BayonetCheck(Vector2 playerPos, Vector2 targetPos)
+		{
+			const float checkRange = 1.5f;
+			if (playerPos.x >= targetPos.x - checkRange  && playerPos.x <= targetPos.x + checkRange)
```

Comment:
You can just use `Vector3.Distance()`

Reply:
Vector3.Distance() (and .magnitude) will make a circular radius around the player where they wont fire their gun and will instead melee (if in range), which is not what I feel would be intuitive here, given a circular radius cant cleanly cover the entirety of all tiles that are adjacent to the player
If I switch this back to use Vector3.Distance() (or .magnitude), then either the outer corners of the tiles directly diagonal to the player will be unmeleeable, or I extend the radius to cover those corners at the cost of having a weird deadzone where you cannot melee because it'd be outa range, but you also cant shoot covering half of the next tile in cardinal directions